### PR TITLE
added negative literals to binary reduce test

### PR DESCRIPTION
**Pull request by TyOverby**
@dotnet/roslyn-compiler 
@AlekseyTs 

As an addendum to https://github.com/dotnet/roslyn/pull/13386

### CODE REVIEW

File: src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs [hunk 1]
```diff
@@ -1965,29 +1965,37 @@ End Module
...
+        Dim b As Integer = &B00100100
+        Dim c As Long = &B001001100110
+
+        Dim d As UShort = &B00100
+        Dim e As UInteger = &B001001100110
+        Dim f As ULong = &B001001100110
+
+        Dim g As Short = -&B101
+        Dim h As Integer = -&B00100100
+        Dim i As Long = -&B001001100110
```
Comment (AlekseyTs) [inline src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs]: These are good. However, what I meant is to test binary literals that themselves represent negative numbers. Also, it looks like we are not testing Byte/SByte.

**Comment by CyrusNajmabadi** [comment]
please tag @dotnet/roslyn-ide for IDE related changes.  Tnx :)

**Comment by jasonmalinowski** [comment]
@dotnet-bot test windows_release_unit32_prtest please and test windows_release_unit64_prtest please. (These are new jobs that are now also required.)

**Comment by TyOverby** [comment]
@AlekseyTs I've added some int-min binary literals to the test cases

File: src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs [hunk 2]
```diff
@@ -1965,29 +1965,59 @@ End Module
...
-        Dim z As Long = &B001001100110
-        Dim x As Long =&B001001100110
+        ' signed
+        Dim a As SByte = &B0111
+        Dim b As Short = &B0101
+        Dim c As Integer = &B00100100
+        Dim d As Long = &B001001100110
+
+        ' unsigned
+        Dim e As UShort = &B01011
```
Comment (AlekseyTs) [inline src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs]: >UShort [](start = 17, length = 6)

Is this supposed to be Byte?

Comment (TyOverby) [inline src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs]: Oh yep.  Done.

**Comment by TyOverby** [comment]
@dotnet-bot test windows_release_unit64_prtest  please

**Comment by TyOverby** [comment]
@AlekseyTs I have fixed the issues in your comments.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by TyOverby** [comment]
@dotnet/roslyn-ide Can you confirm that this is the expected IDE behavior?

**Comment by brettfo** [comment]
:shipit:

**Pull request merged**